cpp/tests/components/scc_test.cu

```diff
@@ -208,4 +218,209 @@ INSTANTIATE_TEST_CASE_P(
     Usecase("test/datasets/cage6.mtx")  // DG "small" enough to meet SCC GPU memory requirements
     ));
 
+struct SCCSmallTest : public ::testing::Test {
+};
+
+TEST_F(SCCSmallTest, CustomGraphSimpleLoops)
+{
+  using IndexT = int;
+
+  size_t nrows = 5;
+  size_t n2    = 2 * nrows * nrows;
+
+  cudaDeviceProp prop;
+  int device = 0;
+  cudaGetDeviceProperties(&prop, device);
+
+  ASSERT_TRUE(n2 < prop.totalGlobalMem);
+
+  // Allocate memory on host
+  std::vector<IndexT> cooRowInd{0, 1, 2, 3, 3, 4};
+  std::vector<IndexT> cooColInd{1, 0, 0, 1, 4, 3};
+  std::vector<IndexT> labels(nrows);
+  std::vector<IndexT> verts(nrows);
+
+  size_t nnz = cooRowInd.size();
+
+  EXPECT_EQ(nnz, cooColInd.size());
+
+  cugraph::GraphCOOView<int, int, float> G_coo(&cooRowInd[0], &cooColInd[0], nullptr, nrows, nnz);
+  auto G_unique                            = cugraph::coo_to_csr(G_coo);
+  cugraph::GraphCSRView<int, int, float> G = G_unique->view();
+
+  rmm::device_vector<IndexT> d_labels(nrows);
+
+  cugraph::connected_components(G, cugraph::cugraph_cc_t::CUGRAPH_STRONG, d_labels.data().get());
+
+  DVector<size_t> d_counts(nrows);
+  auto count_components = get_component_sizes(d_labels.data().get(), nrows, d_counts);
+
+  EXPECT_EQ(count_components, static_cast<size_t>(3));
+
+  std::vector<size_t> v_counts(d_counts.size());
+
+  cudaMemcpy(v_counts.data(),
+             d_counts.data().get(),
+             sizeof(size_t) * v_counts.size(),
+             cudaMemcpyDeviceToHost);
+
+  cudaDeviceSynchronize();
+
+  std::vector<size_t> v_counts_exp{2, 1, 2};
+
+  EXPECT_EQ(v_counts, v_counts_exp);
+}
+
+TEST_F(SCCSmallTest, /*DISABLED_*/ CustomGraphWithSelfLoops)
+{
+  using IndexT = int;
+
+  size_t nrows = 5;
+  size_t n2    = 2 * nrows * nrows;
+
+  cudaDeviceProp prop;
+  int device = 0;
+  cudaGetDeviceProperties(&prop, device);
+
+  ASSERT_TRUE(n2 < prop.totalGlobalMem);
+
+  // Allocate memory on host
+  std::vector<IndexT> cooRowInd{0, 0, 1, 1, 2, 2, 3, 3, 4};
+  std::vector<IndexT> cooColInd{0, 1, 0, 1, 0, 2, 1, 3, 4};
```

Comment:
Thanks for adding the additional tests. I think a FIXME could be added to change these new tests to be parameterized, with input parameters being `nrows`, the COO `col` and `row` indices, and the expected labels and number of components. That would reduce a lot of redundant code and make adding new graphs to test easier.

Reply:
I'm not sure what is the desired course of action here. A `FIXME` saying what? Remember this code path is to be retired (soon). So there's a way bigger FIXME looming here. This would add confusion, I think.